### PR TITLE
[move_group_interface] unable to wait forever for execution capability to come online

### PR DESCRIPTION
**Pull request by davetcoleman**
Fix for recent changes by@rhaschke in https://github.com/ros-planning/moveit/pull/94 

When the default constructor is used, such as in the [moveit_tutorials](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/pr2_tutorials/planning/src/doc/move_group_interface_tutorial.html), the timeout for the capabilities of move_group is should be zero indicating wait forever. For the new execution actionlib interface, it would wait 0 time then throw a runtime error. In this PR I mimicked the behavior of the original ``waitForAction()`` function that the temporary ``waitForExecuteActionOrService()`` function was suppose to have, including renaming a variable to be the same for clarity. 

I also added some helper console feedback.

This fixes our tutorials in Kinetic.

### CODE REVIEW

File: moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp [hunk 1]
```diff
@@ -211,30 +211,52 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     }
   }
 
-  void waitForExecuteActionOrService(ros::WallTime timeout_for_servers)
+  void waitForExecuteActionOrService(ros::WallTime timeout)
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp]: var rename mimics ``void waitForAction(const T &action, const std::string &name, const ros::WallTime &timeout, double allotted_time)``

File: moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp [hunk 2]
```diff
@@ -211,30 +211,52 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
...
                        "\nDeprecation warning: Trajectory execution service is deprecated (was replaced by an action)."
                        "\nReplace 'MoveGroupExecuteService' with 'MoveGroupExecuteTrajectoryAction' in "
                        "move_group.launch");
+      }
       else
+      {
+        ROS_ERROR_STREAM_NAMED("planning_interface", "Unable to find execution action on topic: "
+                               << node_handle_.getNamespace() + move_group::EXECUTE_ACTION_NAME
+                               << " or service: " << node_handle_.getNamespace() + move_group::EXECUTE_SERVICE_NAME);
         throw std::runtime_error("No Trajectory execution capability available.");
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp]: This was previously always being thrown when default MoveGroupInterface constructor used

**Comment by v4hn** [comment]
testing failed due to #311 and because you violated the format :P

**Comment by davetcoleman** [comment]
Format fixed.

**Comment by davetcoleman** [comment]
ping - this is an important bug fix

**Comment by v4hn** [comment]
lgtm.
could you please backport this as required?

**Comment by davetcoleman** [comment]
Done for jade, does not apply for indigo

**Comment by v4hn** [comment]
On Wed, Nov 02, 2016 at 05:37:37PM -0700, Dave Coleman wrote:
> Done for jade, does not apply for indigo

True, but it does apply to the resp. pull-request for indigo that I did not get around to review.

This bug-fix might actually be the fix for the issue I was concerned about there, but I didn't look at the issue in detail.
Sorry, I'm rather busy with teaching these weeks, everyone else can look at that too though. :)

**Pull request merged**